### DISCUSSION TITLE
Are class components supported?

### DISCUSSION THREAD

**Discussion by jamesholcomb** [jpudysz/react-native-unistyles#45] [Q&A] [answered]
Thanks for this excellent lib...The DX has been great.

I have v1.0.0 integrated and working well with functional components and hooks, but have a few legacy class components to deal with.

From the docs, it states _createStyleSheet is interchangeable with StyleSheet.create. You can use objects, and it will function identically to its React Native counterpart._

I tried:

```tsx
const styles = createStyleSheet((theme) => ({
  header: {
    fontWeight: "bold",
    fontSize: theme.fontSizeH2,
    paddingLeft: 15,
    paddingBottom: 15,
    paddingTop: 10
  }
}))
```

but referring to `styles.header` gives:

```
Property 'header' does not exist on type '((theme: AppTheme) => { header: { fontWeight: "bold"; fontSize: number; paddingLeft: number; paddingBottom: number; paddingTop: number; }; }) | { header: { fontWeight: "bold"; fontSize: number; paddingLeft: number; paddingBottom: number; paddingTop: number; }; }'.
  Property 'header' does not exist on type '(theme: AppTheme) => { header: { fontWeight: "bold"; fontSize: number; paddingLeft: number; paddingBottom: number; paddingTop: number; }; }'.
```

**Comment by jpudysz** [comment, marked as answer]
It is interchangeable. However, just as in `StyleSheet.create`, you don't have access to the theme, so it's not available here. What you can do to make it work in a class component is to use it without a function:

```
const styles = createStyleSheet({
  header: {
    fontWeight: "bold",
    // fontSize: theme.fontSizeH2, not available, you need a hook!
    paddingLeft: 15,
    paddingBottom: 15,
    paddingTop: 10
  }
})
```

You cannot access the theme in a class component because you need a hook and context. This will most likely change in version 2.0.

**Reply by jamesholcomb** [reply]
Ah...now I see that `theme` is actually optional.  My legacy class components use `mobx` stores and have access to the unistyles theme via context. Should be good to go.

**Reply by jpudysz** [reply]
Perfect, then it sounds doable 🙏
